### PR TITLE
Port test_rotate in test_functional_tensor.py to pytest

### PR DESCRIPTION
**Pull request by zhiqwang**
Refactored the following tests to pytest in the file test/test/functional_tensor.py mentioned in the issue #3956

- Group D
  - [x] Port `test_rotate`
  - [x] Split into `test_rotate_batch`
  - [x] Split into `test_rotate_deprecation_resample`
  - [x] Split into `test_rotate_interpolation_type`

cc @NicolasHug

### CODE REVIEW

File: test/test_functional_tensor.py [hunk 1]
```diff
@@ -473,7 +471,7 @@ def test_all_ops(self, device, height, width, dt, a, t, s, sh, f, fn):
         # Tolerance : less than 5% (cpu), 6% (cuda) of different pixels
         tol = 0.06 if device == "cuda" else 0.05
         assert ratio_diff_pixels < tol, "{}: {}\n{} vs \n{}".format(
-            (i, a, t, s, sh, f), ratio_diff_pixels, out_tensor[0, :7, :7], out_pil_tensor[0, :7, :7]
+            (NEAREST, a, t, s, sh, f), ratio_diff_pixels, out_tensor[0, :7, :7], out_pil_tensor[0, :7, :7]
```
Comment (zhiqwang) [inline test/test_functional_tensor.py:474]: Seems that the previous PR has a minor bug here, I changed it to `NEAREST`

File: test/test_functional_tensor.py [hunk 2]
```diff
@@ -295,6 +211,88 @@ def test_scale_channel(self):
         assert_equal(scaled_cpu, scaled_cuda.to('cpu'))
 
 
+class TestRotate:
+
+    ALL_DTYPES = [None, torch.float32, torch.float64, torch.float16]
+    scripted_rotate = torch.jit.script(F.rotate)
+    IMG_W = 26
```
Comment (zhiqwang) [inline test/test_functional_tensor.py:218]: I’m not sure if the widths of the two images are equal by coincidence, but under this condition, the `center` can be parametrized.

File: test/test_functional_tensor.py [hunk 3]
```diff
@@ -295,6 +211,88 @@ def test_scale_channel(self):
...
+    @pytest.mark.parametrize('height, width', [(26, IMG_W), (32, IMG_W)])
+    @pytest.mark.parametrize('c', [
+        None,
+        (int(IMG_W * 0.3), int(IMG_W * 0.4)),
+        [int(IMG_W * 0.5), int(IMG_W * 0.6)],
+    ])
+    @pytest.mark.parametrize('dt', ALL_DTYPES)
+    @pytest.mark.parametrize('a', range(-180, 180, 17))
+    @pytest.mark.parametrize('e', [True, False])
+    @pytest.mark.parametrize('f', [None, [0, 0, 0], (1, 2, 3), [255, 255, 255], [1, ], (2.0, )])
```
Comment (NicolasHug) [inline test/test_functional_tensor.py:230]: while we're at it, let's give these better descriptive names: center, angle, expand, fill

**Review by NicolasHug — APPROVED** [review]
nicely done @zhiqwang ! I just have a comment about the naming which we can improve now, but LGTM!

File: test/test_functional_tensor.py (hunk 3 above)
Comment (zhiqwang) [reply test/test_functional_tensor.py]: @NicolasHug Done as suggested.

**Comment by NicolasHug** [comment]
Thanks @zhiqwang ! merging

**Pull request merged**